src/main.js

```diff
@@ -161,13 +171,28 @@ createBookBtn.addEventListener("click", function(event){
 });
 
 function makeBookButton() {
-  currentCover = createCover(userCover.value, userTitle.value, userDecr1.value, userDecr2.value);
-  coverImage.src = currentCover.coverImg; 
-  coverTitle.innerHTML = currentCover.title;
-  coverDescriptor1.innerHTML = currentCover.tagline1;
-  coverDescriptor2.innerHTML = currentCover.tagline2;
-  saveNewInfo();
-  goHome();
+  try {
+    new URL(userCover.value)
+  }
+  catch(error){
+    alert('Please enter a valid url‼️')
+    return
+  }
```

Comment:
Ooh how'd you find this one?  And what's it doing exactly?

Reply:
My mentor suggested to research the try...catch because the if...else statement was really chunky and using some RegEx that was also chunky. So [I read up on it here.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)

Reply:
[This site](https://javascript.info/try-catch) explains it well too!